### DISCUSSION TITLE
Metadata missing DOI

### DISCUSSION THREAD

**Discussion by abhinavdayal** [pymupdf/PyMuPDF#2527] [Q&A]
When we parse metadata of a pdf it ignores certain important keys. See screenshot comparing metadata extracted using PyMuPdf vs that from PyPdf2

![image](https://github.com/pymupdf/PyMuPDF/assets/16099572/a2d225d3-e515-4f93-b03b-d2058e6d26c5)

**Comment by JorjMcKie** [comment]
You are right: The `Document.metadata` property only returns key-values that are part of the official PDF specification.
Another motivation is that PyMuPDF not only supports PDF but half a dozen more document types, which also have their own metadata configurations.
But for PDFs, there are easy ways to get and set arbitrary other, "private" key-value pairs - see [here](https://pymupdf.readthedocs.io/en/latest/recipes-low-level-interfaces.html#how-to-extend-pdf-metadata) in the documentation.
